src/compute_budget.zig

```diff
@@ -0,0 +1,130 @@
+const std = @import("std");
+const sol = @import("solana-program-sdk");
+const bincode = @import("bincode");
+
+const Account = sol.Account;
+
+const ComputeBudget = @This();
+
+pub const id = sol.PublicKey.comptimeFromBase58("ComputeBudget111111111111111111111111111111");
+
+pub fn requestHeapFrame(allocator: std.mem.Allocator, bytes: u32) !sol.Instruction {
+    const data = try bincode.writeAlloc(allocator, ComputeBudget.Instruction{
+        .request_heap_frame = .{ .bytes = bytes },
+    }, .{});
+
+    return sol.Instruction.from(.{
+        .program_id = &id,
+        .accounts = &[_]Account.Param{},
+        .data = data,
+    });
+}
+
+pub fn setComputeUnitLimit(allocator: std.mem.Allocator, units: u32) !sol.Instruction {
+    const data = try bincode.writeAlloc(allocator, ComputeBudget.Instruction{
+        .set_compute_unit_limit = .{ .units = units },
+    }, .{});
+
+    return sol.Instruction.from(.{
+        .program_id = &id,
+        .accounts = &[_]Account.Param{},
+        .data = data,
+    });
+}
+
+pub fn setComputeUnitPrice(allocator: std.mem.Allocator, micro_lamports: u64) !sol.Instruction {
+    const data = try bincode.writeAlloc(allocator, ComputeBudget.Instruction{
+        .set_compute_unit_price = .{ .micro_lamports = micro_lamports },
+    }, .{});
+
+    return sol.Instruction.from(.{
+        .program_id = &id,
+        .accounts = &[_]Account.Param{},
+        .data = data,
+    });
+}
+
+pub fn setLoadedAccountsDataSizeLimit(allocator: std.mem.Allocator, bytes: u32) !sol.Instruction {
+    const data = try bincode.writeAlloc(allocator, ComputeBudget.Instruction{
+        .set_loaded_accounts_data_size_limit = .{ .bytes = bytes },
+    }, .{});
+
+    return sol.Instruction.from(.{
+        .program_id = &id,
+        .accounts = &[_]Account.Param{},
+        .data = data,
+    });
+}
+
+pub const Instruction = union(enum(u8)) {
+    // deprecated variant, reserved value.
+    unused,
+    /// Request a specific transaction-wide program heap region size in bytes.
+    /// The value requested must be a multiple of 1024. This new heap region
+    /// size applies to each program executed in the transaction, including all
+    /// calls to CPIs.
+    request_heap_frame: struct {
+        bytes: u32,
+    },
+    /// Set a specific compute unit limit that the transaction is allowed to consume.
+    set_compute_unit_limit: struct {
+        units: u32,
```

Comment:
nit: can this be renamed to `compute_units` for total clarity?

Reply:
renamed `units` -> `compute_units`

regarding avoiding allocations for program instructions, Ito not blowing up that thread I'd prefer to open the issue under the `solana-program-sdk` and describe a couple of possible allocation strategies:
1. `create` & `destroy`
2. comptime-based
3. etc.

Reply:
Please, that would be perfect